chromium_src/chrome/browser/page_load_metrics/page_load_metrics_initialize.cc

```diff
@@ -0,0 +1,65 @@
+/* Copyright 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#define InitializePageLoadMetricsForWebContents \
+  InitializePageLoadMetricsForWebContents_Chromium
+#include "../../../../../chrome/browser/page_load_metrics/page_load_metrics_initialize.cc"
+#undef InitializePageLoadMetricsForWebContents
+
+// #include "brave/browser/page_load_metrics/brave_page_load_metrics_embedder.h"
+
+#include "brave/components/brave_perf_predictor/browser/buildflags/buildflags.h"
+
+#if BUILDFLAG(ENABLE_BRAVE_PERF_PREDICTOR)
+#include "brave/components/brave_perf_predictor/browser/perf_predictor_page_metrics_observer.h"
+using brave_perf_predictor::PerfPredictorPageMetricsObserver;
+#endif
+
+namespace chrome {
+
+namespace {
+
+class BravePageLoadMetricsEmbedder : public chrome::PageLoadMetricsEmbedder {
+ public:
+  explicit BravePageLoadMetricsEmbedder(content::WebContents* web_contents);
+  ~BravePageLoadMetricsEmbedder() override;
+
+ protected:
+  // page_load_metrics::PageLoadMetricsEmbedderBase:
+  void RegisterEmbedderObservers(
+      ::page_load_metrics::PageLoadTracker* tracker) override;
+
+ private:
+  DISALLOW_COPY_AND_ASSIGN(BravePageLoadMetricsEmbedder);
+};
+
+BravePageLoadMetricsEmbedder::BravePageLoadMetricsEmbedder(
+    content::WebContents* web_contents)
+    : chrome::PageLoadMetricsEmbedder(web_contents) {}
+
+BravePageLoadMetricsEmbedder::~BravePageLoadMetricsEmbedder() = default;
+
+void BravePageLoadMetricsEmbedder::RegisterEmbedderObservers(
+    page_load_metrics::PageLoadTracker* tracker) {
+  PageLoadMetricsEmbedder::RegisterEmbedderObservers(tracker);
+
+#if BUILDFLAG(ENABLE_BRAVE_PERF_PREDICTOR)
+  tracker->AddObserver(std::make_unique<PerfPredictorPageMetricsObserver>());
+#endif
+}
+
+}  // namespace
+
+void InitializePageLoadMetricsForWebContents(
+    content::WebContents* web_contents) {
+  // Change this method? consider to modify the peer in
```

Comment:
I don't think this comment is very useful by itself. If there's something we need to do here can you open a ticket and reference it here?